Sources/LiveViewNative/Modifiers/Input Events Modifiers/DigitalCrownAccessoryModifier.swift

```diff
@@ -0,0 +1,56 @@
+//
+//  DigitalCrownAccessoryModifier.swift
+//  LiveViewNative
+//
+//  Created by Dylan.Ginsburg on 4/26/23.
+//
+
+import SwiftUI
+
+/// Places an accessory View next to the Digital Crown on Apple Watch.
+///
+/// Nested content is referenced by its namespace using the `content` argument.
+///
+/// ```html
+/// <List modifiers={@native |> digital_crown_accessory(content: :dca_view)}>
+///     ...
+///     <digital_crown_accessory:dca_view>
+///         <Image system-name="heart.fill" modifiers={@native |> digital_crown_accessory_visibility(visibility: :visible)} />
+///     </digital_crown_accessory:dca_view>
+/// </List>
+/// ```
+///
+/// ## Arguments
+/// * ``content``
+#if swift(>=5.8)
+@_documentation(visibility: public)
+#endif
+@available(watchOS 9.0, *)
+struct DigitalCrownAccessoryModifier<R: RootRegistry>: ViewModifier, Decodable {
+    @ObservedElement private var element
+    @LiveContext<R> private var context
+    
+    /// A reference to the element that contains the accessory view.
+    #if swift(>=5.8)
+    @_documentation(visibility: public)
+    #endif
+    let content: String
+
+    init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        self.content = try container.decode(String.self, forKey: .content)
+    }
+
+    func body(content: Content) -> some View {
+        content
+            #if os(watchOS)
+            .digitalCrownAccessory {
+                context.buildChildren(of: element, withTagName: self.content, namespace: "digital_crown_accessory")
```

Comment:
This modifier resets the environment, so you need to manually propagate the `coordinatorEnvironment` and `anyLiveContextStorage` environment values to avoid crashing when drawing the content.

Reply:
@carson-katri I copied what looked like similar examples but, tbh, I don't really understand the whole process. Could you help me understand how this is working? If it's too much for text maybe we could screen share and you can walk me through it.

Reply:
The `coordinatorEnvironment` and `anyLiveContextStorage` environment values provide (type-erased) access to the information that's necessary to actually plumb up the live view—primarily a reference to the `LiveViewCoordinator` object, as well as a handful of other things. The various property wrappers, like `@ObservedElement`, rely on getting this information from the environment to function. And so in the handful of places where SwiftUI doesn't propagate the environment, we need to copy these values manually so that the views generated from the child elements have access to them. Right now the two environment keys are bit duplicative and they could probably be merged.

Reply:
The `coordinatorEnvironment and `anyLiveContextStorage` environment values provide (type-erased) access to the information that's necessary to actually plumb up the live view—primarily a reference to the `LiveViewCoordinator` object, as well as a handful of other things. The various property wrappers, like `@ObservedElement`, rely on getting this information from the environment to function. And so in the handful of places where SwiftUI doesn't propagate the environment, we need to copy these values manually so that the views generated from the child elements have access to them. Right now the two environment keys are bit duplicative and they could probably be merged.